pkg/controller/vsphere/session/session.go

```diff
@@ -76,20 +77,27 @@ func GetOrCreate(
 		return nil, fmt.Errorf("error parsing vSphere URL %q", server)
 	}
 
-	soapURL.User = url.UserPassword(username, password)
-
+	// Set user to nil there for prevent login during client creation.
+	// See https://github.com/vmware/govmomi/blob/master/client.go#L91
+	soapURL.User = nil
 	client, err := govmomi.NewClient(ctx, soapURL, insecure)
 	if err != nil {
 		return nil, fmt.Errorf("error setting up new vSphere SOAP client: %w", err)
 	}
 
+	// Set up user agent before login for being able to track mapi component in vcenter sessions list
+	client.UserAgent = "machineAPIvSphereProvider"
+	err = client.Login(ctx, url.UserPassword(username, password))
+	if err != nil {
+		return nil, fmt.Errorf("unable to login to vCenter: %w", err)
+	}
```

Comment:
Nit: I would inline this, very minor though
```suggestion
	if err := client.Login(ctx, url.UserPassword(username, password)); err != nil {
		return nil, fmt.Errorf("unable to login to vCenter: %w", err)
	}
```